src/middleware/ember_html.rs

```diff
@@ -26,11 +67,63 @@ pub async fn serve_html(request: Request, next: Next) -> Response {
         .iter()
         .any(|val| val.to_str().unwrap_or_default().contains("html"))
     {
+        if !matches!(*request.method(), Method::HEAD | Method::GET) {
+            let headers =
+                HeaderMap::from_iter([(header::ALLOW, HeaderValue::from_static("GET,HEAD"))]);
+            return (StatusCode::METHOD_NOT_ALLOWED, headers).into_response();
+        }
+
+        // Come up with an Open Graph image URL. In case a crate page is requested,
+        // we use the crate's name and the OG image base URL from config to
+        // generate one, otherwise we use the fallback image.
+        let og_image_url = 'og: {
+            if let Some(suffix) = path.strip_prefix(PATH_PREFIX_CRATES) {
+                let len = suffix.find('/').unwrap_or(suffix.len());
+                let krate = &suffix[..len];
+
+                // `state.config.og_image_base_url` will always be `Some` as that's required
+                // if `state.config.index_html_template_path` is `Some`, and otherwise this
+                // middleware won't be executed; see `crate::middleware::apply_axum_middleware`.
+                if let Ok(og_img_url) = state.config.og_image_base_url.as_ref().unwrap().join(krate)
+                {
+                    break 'og Cow::from(og_img_url.to_string());
```

Comment:
TIL 😂 

but I'm wondering if this could be simplified somehow. we could extract a `generate_og_image_url()` fn or something like that, and then use `return` instead of the labeled break with return value.

Reply:
Neat right? ;) . But yeah I agree a `fn` would be an improvement